source/Core/Threads/UI/logic/Soldering.cpp

```diff
@@ -9,26 +9,37 @@
 OperatingMode handleSolderingButtons(const ButtonState buttons, guiContext *cxt) {
   if (cxt->scratch_state.state1 >= 2) {
     // Buttons are currently locked
-    switch (buttons) {
-    case BUTTON_F_LONG:
-      if (getSettingValue(SettingsOptions::BoostTemp) && (getSettingValue(SettingsOptions::LockingMode) == lockingMode_t::BOOST)) {
-        cxt->scratch_state.state2 = 1;
+    if (cxt->scratch_state.state1 > 3) {
+      // show locked until timer is up
+      if ((uint16_t)(xTaskGetTickCount() << 2) - (cxt->scratch_state.state1 & ~3) > TICKS_SECOND << 2) {
```

Comment:
Rather than bit packing this I would advise to use a uint32_t or a dedicated uint16_t so that rollover/under wraps and does roughly what is expected. You can use one of the other state variables for the time by itself. If none a spare, add another one :)

I do agree with the concept though, this is good as it keeps it non-blocking which is the end-game goal of the UI code.